lib/logstash/inputs/logstash.rb

```diff
@@ -126,18 +126,12 @@ def inner_http_input_options
   end
 
   def ssl_identity_options
-    # identity_options = {
-    #   'ssl_certificate' => @ssl_certificate || report_invalid_config!('`ssl_certificate` is REQUIRED when `ssl => true`'),
-    #   'ssl_key'         => @ssl_key         || report_invalid_config!('`ssl_key` is REQUIRED when `ssl => true`')
-    # }
-    # identity_options['ssl_key_passphrase'] = @ssl_key_passphrase if @original_params.include?('ssl_key_passphrase')
-
     {}.tap do |identity_options|
       if @ssl_certificate && @ssl_keystore_path
         report_invalid_config!('SSL identity can be configured with EITHER `ssl_certificate` OR `ssl_keystore_*`, but not both')
       elsif @ssl_certificate
         identity_options['ssl_certificate'] = @ssl_certificate
-        identity_options['ssl_key'] = @ssl_key
+        identity_options['ssl_key'] = @ssl_key || report_invalid_config!('`ssl_key` is required when `ssl_certificate` is configured')
```

Comment:
review note: unit tests for the ssl setting will be covered separately with ssl standardization support.